deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/recurrent/LSTMHelpers.java

```diff
@@ -168,6 +170,15 @@ static public FwdPassReturn activateHelper(final Layer layer, final NeuralNetCon
             prevOutputActivations = Nd4j.zeros(new int[] {miniBatchSize, hiddenLayerSize});
         }
 
+        if (helper != null && Nd4j.dataType() != DataBuffer.Type.HALF) {
```

Comment:
We can make the method return null to fall back. So let's do that and log a warning? There are a few other cases like that, so I'll replace all IllegalArgumentException that way. Sounds good?